src/store/people.js

```diff
@@ -17,17 +29,67 @@ export default {
   actions: {
     async list ({ commit }, page = 1) {
       // Fetch People from SWAPI and await the response.
-      const response = await fetch('https://swapi.co/api/people')
-      // Parse the JSON string returned in the response into a JavaScript
-      // object and destructure the results property out of that object.
-      const { results } = await response.json()
-      if (response.ok) {
-        // If the response is OK commit the data to the store.
-        commit('results', results)
-      } else {
-        // If the response is not OK, log the response to the console.
-        console.error(response)
+      const results = await fetch('https://swapi.co/api/people')
+        .then(function (response) {
+          if (response.ok) {
+            return response.json()
+          }
+        })
+      commit('results', results.results)
+    },
+    async getPerson ({ commit }, id) {
+      // Fetch People from SWAPI and await the response.
+      const apiUrl = 'https://swapi.co/api/people/' + id
+
+      const results = await fetch(apiUrl)
+        .then(function (response) {
+          if (response.ok) {
+            return response.json()
+          } else {
+            return null
+          }
+        })
+      if (results === null) {
+        return null
+      }
+      const homeworldUrl = results.homeworld
+      const homeworld = await renderAttributeName(homeworldUrl)
+
+      // saving in this format for easy template rendering
+      results.homeworld = { 'name': homeworld, 'url': homeworldUrl }
+
+      const species = await renderAttributeName(results.species[0])
+      results.species = species
+
+      // Only fetch the vehicle names from the API if the person has vehicles
+      if (results.vehicles !== undefined || results.vehicles.length !== 0) {
+        let vehicles = []
+        for (var i = 0; i < results.vehicles.length; i++) {
+          var vehicleUrl = results.vehicles[i]
+          const vehicleName = await renderAttributeName(vehicleUrl)
+          vehicles[i] = {'name': vehicleName, 'url': vehicleUrl}
+        }
+        results.vehicles = vehicles
+      }
+
+      // Only fetch the starship names from the API if the person has starships
+      if (results.starships !== undefined || results.starships.length !== 0) {
+        let starships = []
+        for (var j = 0; j < results.starships.length; j++) {
+          var starshipUrl = results.starships[j]
+          const starshipName = await renderAttributeName(starshipUrl)
+          starships[j] = {'name': starshipName, 'url': starshipUrl}
+        }
+        results.starships = starships
       }
+
+      commit('results', results)
+      return results
+    }
+  },
+  getters: {
+    getResults: state => {
```

Comment:
This isn't a useful getter since it's only returning the state which can already be retrieved by other methods.